### ISSUE TITLE
linter: `unicorn/prefer-array-flat` reports plain `concat` normalization

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26593]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```json
{
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "rules": {
        "unicorn/prefer-array-flat": "error"
    }
}
```

## Reproduction

```js
export const normalize = (value) => [].concat(value);
export const flatten = (array) => [].concat(...array);
```

## Actual behavior

Oxlint reports both expressions:

```text
Prefer Array#flat() over legacy techniques to flatten arrays.
```

## Expected behavior

Only `[].concat(...array)` should be reported. `[].concat(value)` is plain concat normalization: it passes one concat argument rather than consuming an array of concat arguments.

Current eslint-plugin-unicorn intentionally excludes `[].concat(value)` and `Array.prototype.concat.call([], value)` from `prefer-array-flat`. This changed in [`prefer-array-flat`: Stop reporting plain `concat` normalization (#3106)](https://github.com/sindresorhus/eslint-plugin-unicorn/commit/b4e5ce2cb035685def056ae32e6cfa11ef7d7245). Its documentation assigns ordinary concat modernization to `prefer-spread` instead.

Replacing `[].concat(value)` with `value.flat()` is not equivalent: when `value` is an array containing nested arrays, the former preserves that nesting while the latter removes one level.

**Comment by im10furry** [comment]
Working on this in #26604 — feel free to assign me.

Same root cause also covers `Array.prototype.concat.call([], value)`: upstream's `arrayPrototypeConcat` reports `call` only when the second argument is spread, and `apply` only when it is *not* spread. The PR fixes both plain-concat-normalization families and moves the corresponding upstream-valid cases into the rule's pass list.
